Models/CustomerModel.cs

```diff
@@ -0,0 +1,21 @@
+using DI_Example.Services;
+
+namespace DI_Example
+{
+    public class CustomerModel
+    {
+        public DateTime BirthDate { get; set; }
+
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+
+        private ICustomerService _service;
+
+        public int calculateAge(ICustomerService service)
+        {
+            this._service = service;
```

Comment:
If you must have the _service inside the model, it should be injected not assigned in calculateAge method.
this can cause a lot of issues for any other methods that uses this service, also, it violates SRP. since initiating a global class member is not a responsibility for this method.

Reply:
Umm, yeah Thank you for this clarification i fixed it